data/pigui/modules/system-view-ui.lua

```diff
@@ -408,6 +440,102 @@ function Windows.unexplored.Show()
 	ui.text(lc.UNEXPLORED_SYSTEM_NO_SYSTEM_VIEW)
 end
 
+local function checkSurfPorts(sbody)
+	local children = sbody.children
+	if #children == 0 then return nil end
+	local ports = {}
+	local has_indicators
+	local navTarget = Game.player:GetNavTarget()
+	local selected = systemView:GetSelectedObject().ref
+	for _, child in ipairs(children) do
+		if child.type == "STARPORT_SURFACE" then
+			table.insert(ports, child)
+			local body = child.physicsBody
+			has_indicators = has_indicators or body and body == navTarget or child == selected
+		end
+	end
+	if #ports == 0 then return nil end
+	return ports, has_indicators
+end
+
+local function displaySurfPorts(center, radius, ports, mouseover)
+	local nports = #ports
+	if nports == 0 then return nil end
+	local icon_size = bodyIconSize.x
+	local two_pi = math.pi * 2
+	-- pretty rough estimate
+	local min_rad = icon_size * nports / two_pi
+	radius = math.max(icon_size, min_rad, radius - icon_size / 2)
+
+	local start_ang = 225 * two_pi / 360
```

Comment:
You can also use `math.deg2rad()` here instead of manually doing the degrees->radians conversion.

Reply:
Done!